Add Calico v3.17.3 and v3.18.1

cristicalin opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md and developer guide https://git.k8s.io/community/contributors/devel/development.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:

This PR adds the necessary hashes to support Calico versions 3.17 and 3.18 (while bumping the default to 3.17.3).

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

The release notes for versions 3.17 and 3.18 compared to the aging 3.16 are long so I include just the links here: 
* 3.17: https://docs.projectcalico.org/archive/v3.17/release-notes/
* 3.18: https://docs.projectcalico.org/archive/v3.18/release-notes/

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Upgrade default calico network plugin to version 3.17.3
```

champtar commented:
I'm not following calico closely but why not 3.18.1 ?

cristicalin commented:
My rationale was to add that in a separate PR to keep review effort focused on 1 change at a time. I can add the hashes for 3.18.1 as well here.

One thing that I'm not fully sure is if to go from 3.16.9 to 3.18.1 we would need to go through 3.17.3 as an interim version.

floryut reviewed (approved):
/ok-to-test

Fine to upgrade only to 3.17.3, we could always bump to 3.18.x later on

cristicalin commented:
I have a suspicion that the CI failure is unrelated (markdown job failed in a git fetch), could you re-trigger it?

champtar commented:
Relaunched, kubespray CI always need a bit of love to pass :P

cristicalin commented:
This time the issue was with Calico 3.17 and newer crd files missing the `---` yaml separator. I fixed this and tested both 3.17.3 and 3.18.1 deployments.

It turns out my hash for the 3.18 archive was also wrong so I also fixed it.

@floryut can you bless another CI run?

floryut commented:
> This time the issue was with Calico 3.17 and newer crd files missing the `---` yaml separator. I fixed this and tested both 3.17.3 and 3.18.1 deployments.
> 
> It turns out my hash for the 3.18 archive was also wrong so I also fixed it.
> 
> @floryut can you bless another CI run?

CI is running right now, no error for now 🙏

cristicalin commented:
Again last failure seems unrelated to this change.

champtar commented on roles/network_plugin/calico/tasks/install.yml:
```diff
@@ -119,12 +119,13 @@
         {{ local_release_dir }}/calico-{{ calico_version }}-kdd-crds/*.yaml
       when:
-        - calico_version is version('v3.17.0', '<')
+        - calico_version is version('v3.19.0', '<')
```
Are you sure about this one ?
I think those annotations are not present in 3.17+

cristicalin replied:
Indeed, I looked over the CRDs from 3.17 and 3.18 and they were not there. I just assumed that since there was no other mention of 3.17 before this patch that when condition was an upper bound. That said, the command itself is harmless since it will not change anything in the 3.17+ CRD files but I removed the unnecessary change anyway.

champtar replied:
I agree it's harmless but in kubespray we tend to carry workaround for way longer than needed, once we remove the 3.17 hash we can also remove this task ;)

champtar commented:
/cc @champtar

champtar reviewed (changes requested):
see previous message

champtar commented:
we will have to wait for #7535 to get merged and maybe need rebase

cristicalin commented:
Rebased after https://github.com/kubernetes-sigs/kubespray/pull/7535 was merged as requested.

champtar commented:
/lgtm

champtar commented:
you forgot to update the README
